lms/djangoapps/courseware/access_response.py

```diff
@@ -214,3 +214,25 @@ def __init__(self, partition, user_message=None, user_fragment=None):
         error_code = "no_allowed_user_groups"
         developer_message = u"Group access for {} excludes all students".format(partition.name)
         super(NoAllowedPartitionGroupsError, self).__init__(error_code, developer_message, user_message)
+
+
+class EnrollmentRequiredAccessError(AccessError):
```

Comment:
While we've added these new AccessError classes, they'll only be used by the micro-frontend today.  Other callers to `check_course_access` go through `get_course_with_access` and now use `check_course_access_with_redirect`, which effectively swallows the AccessErrors and turns them into redirects.